blog/2024-01-24-pants-2-19/index.mdx

```diff
@@ -0,0 +1,79 @@
+
+This functionality can be added to any other language with fairly little work. Do you want automatic retries for Scala, Go, or another language Pants supports? Come chat to us in [#development with Slack](https://pantsbuild.slack.com).
+
+### Docker improvements
+
+The [docker backend](https://www.pantsbuild.org/2.19/docs/docker) now has explicit support for using [`buildx` (and thus BuildKit)](https://github.com/docker/buildx#buildx), via [the new `use_buildx` option](https://www.pantsbuild.org/2.19/reference/subsystems/docker#use_buildx). This also allows adding few additional fields for [`docker_image` targets](https://www.pantsbuild.org/2.19/reference/targets/docker_image), for `buildx`-specific functionality:
+
+- [`cache_from` and `cache_to`](https://www.pantsbuild.org/2.19/docs/docker#external-cache-storage-backends) for using external cache storage backends
+- [`output`](https://www.pantsbuild.org/2.19/reference/targets/docker_image#output) to control how/where `buildx` outputs an image when running `pants package ::`
+
+### Improved `parametrize` for tied options
+
+[The `parametrize` helper](https://www.pantsbuild.org/2.19/docs/using-pants/key-concepts/targets-and-build-files#field-default-values#parametrizing-targets) now supports parametrizing multiple fields together, in lock step. For instance:
+
+```python
+# Creates two targets:
+#
+# example:tests@parametrize=py2
+# example:tests@parametrize=py3
+
+python_test(
+    name="tests",
+    source="tests.py",
+    **parametrize("py2", interpreter_constraints=["==2.7.*"], resolve="lock-a"),
+    **parametrize("py3", interpreter_constraints=[">=3.6,<3.7"], resolve="lock-b"),
+)
+```
+
+In addition, `parametrize` can now be used with a single value (`field=parametrize("for-consistency")`), and now works on target generations in more cases (particularly those with "plugin" fields).
+
+### Other noteworthy changes
+
+- New versions of [the `get-pants.sh` script](https://www.pantsbuild.org/2.19/docs/getting-started/installing-pants) install to `~/.local/bin`, not `~/bin`. Remember to update your CI pipelines!
+- Environment variables that aren't valid UTF-8 no longer cause Pants to crash (note, `PANTS_...` environment variables that Pants needs to read but aren't UTF-8 will be ignored).
+- Reference documentation for subsystem options includes how to set them in `pants.toml`, in addition to the environment variable and CLI arguments.
+- The `.pids/` internal directory is now under `.pants.d/` by default, rather that adjacent to it (this can be controlled via [the `[GLOBAL].pants_subprocessdir` option](https://www.pantsbuild.org/2.19/reference/global-options#pants_subprocessdir)).
```

Comment:
From memory, other release blog posts will sometimes end with a repetition of the call to actions, etc, and/or some brief thanks. What do you think about doing that here too?